qt/widgets/common/src/FunctionTreeView.cpp

```diff
@@ -1220,6 +1228,8 @@ void FunctionTreeView::addFunctionEnd(int result) {
       return;
   }
   emit functionAdded(QString::fromStdString(f->asString()));
+  setGlobalParameters(globalParameters);
+  emit globalsChanged(globalParameters);
```

Comment:
What is the difference between using `emit` and just calling the function?

Reply:
globalsChanged is marked as a qt signal. If we emit this signal anything connected to it (e.g the presenter) will be notified. This is necessary as changing this value in the view should notify the presenter, which will then update the model with the new global parameters.